src/caffe/proto/caffe.proto

```diff
@@ -445,6 +450,40 @@ message DataParameter {
   optional bool mirror = 6 [default = false];
 }
 
+// Message that stores parameters used by DataLayer
+message DatumDBParameter {
+  enum Mode {
+    NEW = 0;
+    WRITE = 1;
+    READ = 2;
+  }
+  // Specify the datumDB Backend.
+  optional string backend = 1 [default = "leveldb"];
+  // Specify the datumDB mode.
+  optional Mode mode = 2 [default = READ];
+  // Specify the datumDB source.
+  optional string source = 3;
+  // If true it will loop around the datumDB when needed
+  optional bool loop = 4 [default = true];
```

Comment:
This is only part of the solution to avoid looping during test, the other part will be handle the case of the last batch by the prefetch thread and data_layer.